bc_obps/compliance/tests/service/elicensing/test_elicensing_obligation_service.py

```diff
@@ -734,3 +776,104 @@ def test_get_obligations_for_reminders(self):
+        will_send_2 = _make_obligation_for(cp, due_date=date(2025, 11, 30))  # == deadline
+        filtered_out = _make_obligation_for(cp, due_date=date(2025, 12, 1))  # > deadline
+
+        # Return an actual QuerySet so the method’s extra .filter(...) works
+        base_qs = ComplianceObligation.objects.filter(pk__in=[will_send_1.pk, will_send_2.pk, filtered_out.pk])
+        mock_get_current_reporting_year.return_value = ry
+        mock_get_obligations_for_reminders.return_value = base_qs
+
+        # Act
+        ElicensingObligationService.send_notice_for_penalty_accrual_for_current_period()
+
+        # Assert base calls
+        mock_get_current_reporting_year.assert_called_once()
+        mock_get_obligations_for_reminders.assert_called_once_with(cp)
+
+        # Assert only <= deadline are enqueued
+        called_ids = {c.args[0] for c in mock_retryable_penalty_accrual_execute.call_args_list}
+        assert called_ids == {will_send_1.id, will_send_2.id}
+
+    def test_send_notice_for_penalty_accrual_for_current_period__no_compliance_period_found(
+        self,
+        mock_retryable_penalty_accrual_execute,
+        mock_get_obligations_for_reminders,
+        mock_get_current_reporting_year,
+    ):
+        """
+        - No CP for current RY -> return early, no queries / no emails.
+        """
+        ry = ReportingYear.objects.get(reporting_year=2024)
+        mock_get_current_reporting_year.return_value = ry
+
+        # Ensure there is no CP for this reporting year
+        CompliancePeriod.objects.filter(reporting_year=ry).delete()
+
+        # Act
+        ElicensingObligationService.send_notice_for_penalty_accrual_for_current_period()
+
+        # Assert
+        mock_get_current_reporting_year.assert_called_once()
+        mock_get_obligations_for_reminders.assert_not_called()
+        mock_retryable_penalty_accrual_execute.assert_not_called()
+
+    def test_send_notice_for_penalty_accrual_for_current_period__no_obligations_after_deadline_filter(
+        self,
+        mock_retryable_penalty_accrual_execute,
+        mock_get_obligations_for_reminders,
+        mock_get_current_reporting_year,
+    ):
+        """
+        - CP exists; base set returned by _get_obligations_for_reminders
+        - All obligations have due_date AFTER the business deadline
+        - Result: no emails enqueued
+        """
+        ry = ReportingYear.objects.get(reporting_year=2024)
+        cp, _ = CompliancePeriod.objects.get_or_create(
+            reporting_year=ry,
+            defaults={
+                "invoice_generation_date": date(2024, 11, 1),
+                "compliance_deadline": date(2025, 11, 30),
+            },
+        )
+
+        # Build obligations that match base filters but fail the deadline filter
+        o1 = _make_obligation_for(cp, due_date=date(2025, 12, 1))
+        o2 = _make_obligation_for(cp, due_date=date(2026, 1, 5))
```

Comment:
suggest adding a mock obligation for a previous year to make sure we're only sending out reminders for the current year